### ISSUE TITLE
Question/Bug regarding versioning on openupm compared to github

### ISSUE THREAD

**Issue by indrif** [openupm/openupm#4593]
## Bug report

I am currently in the process of releasing one of our SDKs to openupm. We have successfully set up an open source repository along with pipelines etc to create new tags/releases etc. We recently tried to deploy our first version. In our repository the tags and releases have been created successfully with 1.0.0 however on openupm it says the version deployed is 0.0.1:

Our github repository:
![image](https://github.com/openupm/openupm/assets/6198331/38ce9e5d-24be-4569-a297-62c83c37281d)

Openupm for our package:
![image](https://github.com/openupm/openupm/assets/6198331/cda0256d-aac6-4b95-9e5b-148bebb01a13)

Is this a bug or does openupm do some calculations on its own?

#### Bug category

- [ ] Website: issues related to browse or discover packages.
- [x] Pipelines: issues related to publish or manage a package.
- [x] Registry: issues related to the registry server.
- [ ] Unity Editor: issues related to how OpenUPM works with the Unity Editor.
- [ ] Documentation
- [ ] Others

#### Steps to reproduce

1. Deploy version 1.0.0 in a lib

#### What is expected?

Version should be 1.0.0 on Openupm

#### What is actually happening?

Version is 0.0.1

#### Other relevant information

Repository: https://github.com/appylar/published-unity-sdk

**Comment by favoyang** [comment]
Because of the package.json version of tag `upm/1.0.0` is `0.0.1`: https://github.com/appylar/published-unity-sdk/blob/b942b0100ba0c6fa500c8f91738cf0d4b974fc0a/package.json

This will be treated as a failure by build-pipeline in the future (refs https://github.com/openupm/openupm/issues/4483). But for now, we let it pass and published `0.0.1`.

You can bump the version to 1.0.1, then create Git tag 1.0.1 to fix this.

I checked your ci.yml (https://github.com/appylar/published-unity-sdk/blob/main/.github/workflows/ci.yml), you're using semantic-release-action, it uses git tags to detect the initial version. That may cause the first released version mismatch. Try to generate a new commit to trigger another build, then verify the package.json version with the Git tag. It could be an empty commit like:

```
git commit --allow-empty -m "fix: rebuild package.json version"
git push
```

BTW, the split upm branch is no longer needed by our build-pipeline which supports detecting the package.json path automatically.

**Comment by indrif** [comment]
Thanks for the very detailed explanation! I will fix the issue. 

Regarding the upm split branch thing, this was a suggestion from the medium article linked from your documentation. I also noticed that a lot of the content in that article was very outdated. Would be nice with a new tutorial on this for future users!

Anyway, thanks! Have a great day :)
